tests/components/honeywell/test_climate.py

```diff
@@ -0,0 +1,1025 @@
+) -> None:
+    """Test controlling the auxheat through service calls."""
+    await init_integration(hass, config_entry)
+    entity_id = f"climate.{device.name}"
+
+    await hass.services.async_call(
+        CLIMATE_DOMAIN,
+        SERVICE_SET_AUX_HEAT,
+        {ATTR_ENTITY_ID: entity_id, ATTR_AUX_HEAT: True},
+        blocking=True,
+    )
+    device.set_system_mode.assert_called_once_with("emheat")
+
+    device.set_system_mode.reset_mock()
+    await hass.services.async_call(
+        CLIMATE_DOMAIN,
+        SERVICE_SET_AUX_HEAT,
+        {ATTR_ENTITY_ID: entity_id, ATTR_AUX_HEAT: False},
+        blocking=True,
+    )
+    device.set_system_mode.assert_called_once_with("heat")
+
+
+async def test_fan_modes_service_calls(
+    hass: HomeAssistant, device: MagicMock, config_entry: MagicMock
+) -> None:
+    """Test controlling the fan modes through service calls."""
+    await init_integration(hass, config_entry)
+    entity_id = f"climate.{device.name}"
+
+    await hass.services.async_call(
+        CLIMATE_DOMAIN,
+        SERVICE_SET_FAN_MODE,
+        {ATTR_ENTITY_ID: entity_id, ATTR_FAN_MODE: FAN_AUTO},
+        blocking=True,
+    )
+
+    device.set_fan_mode.assert_called_once_with("auto")
+
+    device.set_fan_mode.reset_mock()
+    await hass.services.async_call(
+        CLIMATE_DOMAIN,
+        SERVICE_SET_FAN_MODE,
+        {ATTR_ENTITY_ID: entity_id, ATTR_FAN_MODE: FAN_ON},
+        blocking=True,
+    )
+
+    device.set_fan_mode.assert_called_once_with("on")
+
+    device.set_fan_mode.reset_mock()
+    await hass.services.async_call(
+        CLIMATE_DOMAIN,
+        SERVICE_SET_FAN_MODE,
+        {ATTR_ENTITY_ID: entity_id, ATTR_FAN_MODE: FAN_DIFFUSE},
+        blocking=True,
+    )
+
+    device.set_fan_mode.assert_called_once_with("circulate")
+
+
+async def test_service_calls_off_mode(
+    hass: HomeAssistant, device: MagicMock, config_entry: MagicMock
+) -> None:
+    """Test controlling the entity through service calls."""
+
+    device.system_mode = "off"
+
+    await init_integration(hass, config_entry)
+    entity_id = f"climate.{device.name}"
+
+    async_fire_time_changed(
+        hass,
+        utcnow() + SCAN_INTERVAL,
+    )
+    await hass.async_block_till_done()
+
+    await hass.services.async_call(
+        CLIMATE_DOMAIN,
+        SERVICE_SET_TEMPERATURE,
+        {ATTR_ENTITY_ID: entity_id, ATTR_TEMPERATURE: 15},
+        blocking=True,
+    )
+
+    await hass.services.async_call(
+        CLIMATE_DOMAIN,
+        SERVICE_SET_TEMPERATURE,
+        {
+            ATTR_ENTITY_ID: entity_id,
+            ATTR_TARGET_TEMP_LOW: 25.0,
+            ATTR_TARGET_TEMP_HIGH: 35.0,
+        },
+        blocking=True,
+    )
+    device.set_setpoint_cool.assert_called_with(95)
+    device.set_setpoint_heat.assert_called_with(77)
+
+    device.set_setpoint_heat.reset_mock()
+    device.set_setpoint_heat.side_effect = aiosomecomfort.SomeComfortError
```

Comment:
Add a comment to explain the purpose / expected outcome of making `set_setpoint_heat` raiss, same comment in the other cases

Reply:
Looks like I missed the actual outcome of setting the temperature and retruning the SomeComfortError - in this case it's a log entry only.
I'll need to do a bit of research on testing the log entries for these

Reply:
Reviewing this work, I found several changes required due to code updates after the tests were written.
I believe I've covered the questions you had by performing a test rather than adding a comment.  It was good for you to question the specifics, as several of them were not testing anything :)